core/src/epicli/data/common/ansible/playbooks/roles/repository/files/download-requirements/common/common_functions.sh

```diff
@@ -0,0 +1,238 @@
+#
+# :param $1: repository to be updated
+# :return: apt return value
+#
+    last_error=$(apt update -o Dir::Etc::sourcelist=$1 2>&1 >/dev/null)
+    return $?
+}
+
+
+__check_apt() {
+#
+# Use `apt update` to make sure that there is connection to repositories.
+#
+# :param $@: repos to be tested
+# :return: 0 - success, 1 - failure
+#
+    echol "Testing apt connection:"
+
+    (( $# > 0 )) || exit_with_error "__check_apt: no repositories provided"
+    local repos=$@
+
+    (( $UID == 0 )) || exit_with_error "apt needs to be run as a root"
+
+    __at_least_one_test_pass __test_apt_repo $repos
+    return $?
+}
+
+
+__test_yum_repo() {
+#
+# List packages from a single repository.
+#
+# :param $1: repository to be listed
+# :return: yum return value
+#
+    last_error=$(yum --quiet --disablerepo=* --enablerepo=$1 list available 2>&1 >/dev/null)
+    return $?
+}
+
+
+__check_yum() {
+#
+# Use `yum list` to make sure that there is connection to repositories.
+# Query available packages for each repository.
+#
+# :param $@: repositories to be tested
+# :return: 0 - success, 1 - failure
+#
+    echol "Testing yum connection:"
+
+    (( $# > 0 )) || exit_with_error "__check_yum: no repositories provided"
+    local repos=$@
+
+    __at_least_one_test_pass __test_yum_repo $repos
+    return $?
+}
+
+
+__test_crane_repo() {
+#
+# List packages from a single repository.
+# Requires $CRANE_BIN to be defined
+#
+# :param $1: repository to be listed
+# :return: crane return value
+#
+    last_error=$($CRANE_BIN ls $1 2>&1 >/dev/null)
+    return $?
+}
+
+
+__check_crane() {
+#
+# Use `crane ls` to make sure that there is connection to repositories.
+# Query available packages for each repository.
+#
+# :param $@: repositories to be tested
+# :return: 0 - success, 1 - failure
+#
+    echol "Testing crane connection:"
+
+    (( $# > 0 )) || exit_with_error "__check_crane: no repository provided"
+    local repos=$@
+
+    __at_least_one_test_pass __test_crane_repo $repos
+    return $?
+}
+
+
+# Tools which can be tested:
+declare -A tools=(
+[curl]=__check_curl
+[wget]=__check_wget
+[apt]=__check_apt
+[yum]=__check_yum
+[crane]=__check_crane
+)
+
+
+check_connection() {
+#
+# Run connection test for target `tool` with optional `url` parameter.
+# Requires $internet_access_checks_enabled to be defined.
+#
+# :param $1: which `tool` to test
+# :param $@: optional parameters used by some tools such as `url`
+#
+    [[ $internet_access_checks_enabled == "no" ]] && return 0
+
+    [[ $# -lt 1 ]] && exit_with_error "(tool) argument not provided"
+    local tool=$1
+
+    shift  # discard tool variable
+
+    [[ ! -n ${tools[$tool]} ]] && exit_with_error "no such tool ($tool)"
+
+    (  # disable -e in order to handle non-zero return values
+        set +e
+
+        ${tools[$tool]} $@
+
+        if (( $? == 0 )); then
+            echol "Connection successful."
+        else
+            exit_with_error "Connection failure, reason: ($last_error)"
```

Comment:
```suggestion
            exit_with_error "Connection failure, reason: [$last_error]"
```
() are probably used more often than [] in error message. 
IMO `[curl: (6) Could not resolve host: google.com; Name or service not known]` looks better than `(curl: (6) Could not resolve host: google.com; Name or service not known)`